windows/DateTimePickerWindows/DatePickerModuleWindows.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "DatePickerModuleWindows.h"
+
+#include <winrt/Microsoft.ReactNative.Xaml.h>
+#include <winrt/Microsoft.UI.Xaml.h>
+
+namespace winrt::DateTimePicker {
+
+void DatePickerModule::Initialize(winrt::Microsoft::ReactNative::ReactContext const &reactContext) noexcept {
+  m_reactContext = reactContext;
+}
+
+void DatePickerModule::Open(ReactNativeSpecs::DatePickerModuleWindowsSpec_DatePickerOpenParams &&params,
+                            winrt::Microsoft::ReactNative::ReactPromise<ReactNativeSpecs::DatePickerModuleWindowsSpec_DatePickerResult> promise) noexcept {
+  // Ensure XAML is initialized
+  winrt::Microsoft::ReactNative::Xaml::implementation::XamlApplication::EnsureCreated();
+
+  // Clean up any existing picker
+  CleanupPicker();
+
+  // Store the promise
+  m_currentPromise = promise;
+
+  // Store timezone offset
+  if (params.timeZoneOffsetInSeconds.has_value()) {
+    m_timeZoneOffsetInSeconds = static_cast<int64_t>(params.timeZoneOffsetInSeconds.value());
+  } else {
+    m_timeZoneOffsetInSeconds = 0;
+  }
+
+  // Create the CalendarDatePicker
+  m_datePickerControl = winrt::Microsoft::UI::Xaml::Controls::CalendarDatePicker{};
+
+  // Set properties from params
+  if (params.dayOfWeekFormat.has_value()) {
+    m_datePickerControl.DayOfWeekFormat(winrt::to_hstring(params.dayOfWeekFormat.value()));
+  }
```

Comment:
```suggestion
  if (params.dayOfWeekFormat) {
    m_datePickerControl.DayOfWeekFormat(winrt::to_hstring(*params.dayOfWeekFormat));
  }
```

`std::optional` has a boolean context which makes it brief and clean.  Refer https://en.cppreference.com/w/cpp/utility/optional/operator_bool.html.

Remember [The Best Code is No Code At All](https://blog.codinghorror.com/the-best-code-is-no-code-at-all/).